src/solver/mlir_common.cpp

```diff
@@ -41,6 +43,30 @@ std::string InsertGToLayout(const std::string& layout, char dim)
     return layout_with_g.insert(index, 1, 'G');
 }
 
+const char* DTypeName(miopenDataType_t ty) {
```

Comment:
I know in all scenarios you are returning a string literal with static storage duration. However I'd prefer this to be returning a string so client of this function never worry about the implementation details of this one.

Reply:
On the other hand, if we're ever using anything but string literals, we can refactor, and this has the advantage of making it clear that it's meant to be a static string literal.

Reply:
Fine by me.